datafusion/core/src/physical_optimizer/sort_enforcement.rs

```diff
@@ -1220,12 +1220,10 @@ mod tests {
             sort_expr("non_nullable_col", &schema),
         ];
         let repartition_exec = repartition_exec(spm);
-        let sort2 = Arc::new(SortExec::new_with_partitioning(
-            sort_exprs.clone(),
-            repartition_exec,
-            true,
-            None,
-        )) as _;
+        let sort2 = Arc::new(
+            SortExec::new(sort_exprs.clone(), repartition_exec, None)
+                .with_preserve_partitioning(true),
```

Comment:
I think this makes it clearer that the sort is being constructed to  preserve the input partitioning